spec/style-rules.md

```diff
@@ -4,28 +4,50 @@
 
 * [Definitions](#definitions)
   * [Current Style Rule](#current-style-rule)
+  * [Current Keyframe Block](#current-keyframe-block)
 * [Semantics](#semantics)
 
 ## Definitions
 
 ### Current Style Rule
 
 The *current style rule* is the CSS style rule that was created by the innermost
-[execution of a style rule](#semantics).
+[execution of a style rule](#semantics), `@media` rule, `@supports` rule, or
+unknown at-rule.
+
+### Current Keyframe Block
+
+The *current keyframe block* is the CSS keyframe block that was created by the
+innermost [execution of a style rule](#semantics).
 
 ## Semantics
 
 To execute a style rule `rule`:
 
-* Let `selector` be the result of evaluating all interpolation in `rule`'s
-  selector and parsing the result as a selector list.
+* Let `selector-text` be the result of evaluating all interpolation in `rule`'s
+  selector.
 
-* Let `parent` be the [current style rule] or current at-rule if one exists, or
-  the innermost if both exist.
+* Let `parent` be the [current style rule], at-rule, or keyframe block if one
+  exists, or the innermost if multiple exist.
 
   [current style rule]: #current-style-rule
 
-* If `parent` is a style rule whose stylesheet wasn't [parsed as CSS]:
+* If `parent` is a keyframe block, throw an error.
+
+* Otherwise, if `parent` is an unknown at-rule whose name without vendor
+  prefixes is "keyframes":
+
+  * Let `selector` be the result of parsing `selector-text` as a keyframe
+    selector.
+
+  * Append a keyframe block with selector `selector` to `parent`.
+
+  * Evaluate each child of `rule`.
+
+  * Cease evaluating `rule`.
```

Comment:
I'm kinda confused by the last bullet. What does it mean by "cease evaluating" and why is it not implied that `rule` will stop being evaluated when all children are evaluated? 

Or does this line means that the next bullet point in line 49 should be ignored?

Reply:
Yeah, it's intended to be the spec equivalent of `return;`. I'm not thrilled with the wording either... if you have a better suggestion I'm very interested.

Reply:
Oh I see, I was confused partly because lines 75 and 77 look similar (two "otherwise" bullets chained together) but they didn't have the "cease evaluating" text. I see now that that's because in _that_ case the evaluation keeps going into line 81, and in _this_ case the evaluation stops completely.

I think this is fine, thanks for the clarification :)